app/models/health/epic_thrive.rb

```diff
@@ -151,10 +151,16 @@ def update_thrive_assessment!
       assessment.decline_to_answer = true if !@any_answer && @any_decline
 
       assessment.save!
-      patient.hrsn_screenings.where(instrument_id: assessment.id).first_or_create(
-        instrument: assessment,
-        created_at: assessment.created_at,
-      )
+      # A thrive collected outside of the enrollment window may throw an exception if it
+      # creates an HRSN QA, which we will ignore
+      begin
+        patient.hrsn_screenings.where(instrument_id: assessment.id).first_or_create(
+          instrument: assessment,
+          created_at: assessment.created_at,
+        )
+      rescue ActiveRecord::RecordInvalid
```

Comment:
This seems a bit like a sledge hammer.  Is there any way to validate that it's the date that's causing the error?